src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,209 @@
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
+ * `enveloping cube` and `enveloping sphere` refer to the bounding surfaces of
+ * the outer Sphere of ten wedges making up the last layer of the domain.
+ *
+ * The plane along which the domain divides into two hemispheres is referred
+ * to as the "cutting plane".
+ */
+
+template <typename TargetFrame>
+class BinaryCompactObject : public DomainCreator<3, TargetFrame> {
+ public:
+  struct InnerRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Inner coordinate radius of Layer 1 for Object A."};
+  };
+
+  struct OuterRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Outer coordinate radius of Layer 1 for Object A."};
+  };
+
+  struct XCoordObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of center for Object A."};
+  };
+
+  struct ExciseInteriorA {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct InnerRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Inner coordinate radius of Layer 1 for Object B."};
+  };
+
+  struct OuterRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Outer coordinate radius of Layer 1 for Object B."};
+  };
+
+  struct XCoordObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of the center for Object B."};
+  };
+
+  struct ExciseInteriorB {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct RadiusOuterCube {
+    using type = double;
+    static constexpr OptionString help = {
+        "Radius of Layer 3 which circumscribes the Frustums."};
+  };
+
+  struct RadiusOuterSphere {
+    using type = double;
+    static constexpr OptionString help = {"Radius of the entire domain."};
+  };
+
+  struct InitialRefinement {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial refinement level. Applied to each dimension."};
+  };
+
+  struct InitialGridPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial number of grid points in each dim per element."};
+  };
+
+  struct UseEquiangularMap {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Use equiangular instead of equidistant coordinates."};
+    static type default_value() { return true; }
+  };
+
+  struct OriginPreimage {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of center of mass before translation."};
+    static type default_value() { return 0.0; }
+  };
```

Comment:
It seems as though the `XCoordObjectA`, `XCoordObjectB` and `OriginPreimage` are all fairly indirect ways of specifying the domain from our way of thinking about it. We'd find it more intuitive to just specify the locations, in the final coordinate system, of the two compact object centers. Then, there doesn't have to be an additional shift associated with `OriginPreimage` specified by the options (the implementation can still do what it needs to with intermediate variables). Is there a reason that understanding is incorrect?

Reply:
@moxcodes It is possible to set up the options such that the final coordinates are specified, but I'll need to add restrictions on the possible values. For example, `XCoordObjectA = 5` and `XCoordObjectB = 10` will not be a sensible choice since the axis of rotation is located at x=0. The removal of the OriginPreimage option means I'm leaving it up to the user to make sure that the choices for the locations of the compact objects leave the center of mass located at x=0 as well. For example, if `XCoordObjectA = 5` and `XCoordObjectB = 5`, there is no additional option for moving the CoM to the origin if one is evolving an unequal mass system - it'll be up to the user to specify coordinates that make sense. I can add these details to the documentation but I wanted to make sure that this is what we want. Perhaps something like `In the final coordinates, the rotation axis is located at x=0, so please make sure that XCoordObjectA and XCoordObjectB, as well as the resulting location of the center of mass are compatible with this.`

Reply:
Hmm, this is still a bit confusing to me then. Is it true that given a particular pair of `XCoordObjectA` and `XCoordObjectB`, there is exactly one reasonable choice for `OriginPreimage`? If so, I think the code should just compute that single reasonable choice and include in the documentation the statement of what is being done.

But, if it's true that there is only one reasonable choice for `OriginPreimage` given the original pair of coordinates of A and B, then it seems as though there's actually only one useful degree of freedom here : the coordinate distance between the two centers.

Reply:
@moxcodes I have removed `OriginPreimage` in my latest commits, but it is not true that a unique, "correct" value for `OriginPreimage` can be computed given an arbitrary `XCoordObjectA` and `XCoordObjectB`. There is no information as to the location of the center of mass/ where the axis of rotation should be between the two centers, and this is the role that was filled by `OriginPreimage`. (With `OriginPreimage` the x-coordinate of the CoM in the pre-mapped coordinates would be specified and the domain would be constructed such that the CoM is then mapped to the origin, so that rotation about the z-axis keeps the CoM in place). With the `OriginPreimage` option removed, I am now assuming that the user has gone through the effort of making sure the CoM is already at the origin when specifying `XCoordObjectA` and `XCoordObjectB`, which are now the locations of the centers of the compact objects in the final coordinates. Is this more in line with what you were thinking?

Reply:
Oh, I see -- the domain creator doesn't get access to the mass, so it can't make the remaining decision on its own.
Yeah, I think this is substantially more intuitive; it's definitely what @markscheel and I were thinking when we first reviewed it. Thanks!